File: apis/src/main/java/com/beat/global/external/s3/api/dto/PerformanceMakerPresignedUrlFindAllResponse.java
package com.beat.global.external.s3.api.dto;

import com.beat.contracts.storage.PerformancePresignedUrls;
import java.util.Map;

public record PerformanceMakerPresignedUrlFindAllResponse(
	Map<String, Map<String, String>> performanceMakerPresignedUrls
) {

	public static PerformanceMakerPresignedUrlFindAllResponse from(PerformancePresignedUrls response) {
		return new PerformanceMakerPresignedUrlFindAllResponse(response.performanceMakerPresignedUrls());
	}
}

File: apis/src/main/kotlin/com/beat/apis/file/api/response/PerformanceMakerPresignedUrlFindAllResponse.kt
package com.beat.apis.file.api.response

import com.beat.contracts.storage.ImagePresignedUpload

@ConsistentCopyVisibility
data class PerformanceMakerPresignedUrlFindAllResponse private constructor(
    val performanceMakerPresignedUrls: Map<String, Map<String, String>>,
    val performanceMakerPresignedUploads: Map<String, Map<String, ImagePresignedUpload>>,
) {
    companion object {
        fun from(
            performanceMakerPresignedUploads: Map<String, Map<String, ImagePresignedUpload>>,
        ): PerformanceMakerPresignedUrlFindAllResponse {
            val legacyUrls = performanceMakerPresignedUploads.mapValues { (_, uploads) ->
                uploads.mapValues { (_, upload) -> upload.uploadUrl }
            }
            return PerformanceMakerPresignedUrlFindAllResponse(legacyUrls, performanceMakerPresignedUploads)
        }
    }
}
